### PR TITLE
Refactor Button + add underline prop

### PR DESCRIPTION
**Pull request by craig-mulligan**
I was reading the styled-components docs, and they suggest using StyledComponent.extend`ccs` rather than styled(Component)`css` to extend styles. It has performance improvements and only renders the differing styles rather than creating new classes for each type of component.

Connects to #59
Change-Type: patch

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -44,79 +71,39 @@ const Base = styled.button`
...
-    color: '#fff',
-    bg: get(theme.colors[type], 'main'),
-    active: {
-      color: '#fff',
-      bg: get(theme.colors[type], 'dark')
-    }
+  &:hover,
+  &:focus,
+  &:active {
+    background: none;
```
Comment (thgreasi) [inline src/components/Button.js]: Just a note, in the billing page of the dashboard the underline buttons get filled on hover/focus/active, looking similar with the hovered outlined buttons. Never had a spec on this and neither have a strong opinion though.

File: src/components/Button.js [hunk 2]
```diff
@@ -44,79 +71,39 @@ const Base = styled.button`
...
+    color: ${props =>
+    getColor(props, 'color', 'main') || props.theme.colors.text.main};
+    border-bottom: 2px solid;
```
[the remark this answers is not in the document]
Comment (craig-mulligan) [reply src/components/Button.js]: Nice 👍

File: src/components/Button.js (hunk 1 above)
Comment (craig-mulligan) [reply src/components/Button.js]: Hmm, yea I'm not sure either? I was basing it off the designs for the marketing site pricing page. Great that we are catching these inconsistencies :) @konmouz any opinions? If not we can create an issue and change the button when we have a design for it.

Comment (konmouz) [reply src/components/Button.js]: @craig-mulligan @thgreasi, at the moment @taahirisaacs is preparing a new styleguide version that will be shared soon. After that, we can review what elements are missing and populate the guide.

Comment (craig-mulligan) [reply src/components/Button.js]: Okay thanks @konmouz, I'll create an issue so we can revisit it once we have designs.

**Comment by craig-mulligan** [comment]
@thgreasi @LucianBuzzo could you have a sec. 👍

**Pull request merged**